pkg/apis/kops/componentconfig.go

```diff
@@ -292,6 +292,29 @@ type KubeControllerManagerConfig struct {
 	UseServiceAccountCredentials *bool `json:"useServiceAccountCredentials,omitempty" flag:"use-service-account-credentials"`
 }
 
+// CloudControllerManagerConfig is the configuration for the controller
+type CloudControllerManagerConfig struct {
```

Comment:
If we need a full struct, why is this not in the ClusterSpec.  @andrewsykim / @justinsb can I get some comments here. Do we need another full struct for the `CloudControllerManagerConfig`.  Where in our API should it live?

Reply:
I believe Sid added this to ClusterSpec eventually :). I think we're better off treating this as a cluster addon as opposed to adding this to cluster spec.

For what it's worth, I don't think we need a full struct for `CloudControllerManager` since we can actually copy most of the flags from `KubeControllerManager`, but this is probably better for readability/visibility.

Reply:
@andrewsykim @chrislovecnm I did add this to ClusterSpec eventually. This is a core component i.e. without this, service creation won't happen, cloudroutes won't be setup etc.. Semantically, this seems more like a core component than a addon. That's why I added this alongside kubeControllerManagerConfig.

CCM has fewer flags than KCM, and I can see it evolving independent of the KCM. For eg. we will add Volume support for CCM, but we won't add those options to KCM. Having a separate structure facilitatates that.